ui/src/utils/decimals.test.ts

```diff
@@ -0,0 +1,78 @@
+import { amountToDecimal, decimalToAmount } from './decimals';
+
+test('amountToDecimal() should be padded by correct number of decimals', () => {
+  const noDecimal = '123456789';
+  expect(amountToDecimal(noDecimal, undefined)).toEqual(undefined);
+  expect(amountToDecimal(noDecimal, 0)).toEqual('123456789');
+  expect(amountToDecimal(noDecimal, 1)).toEqual('1234567890');
+  expect(amountToDecimal(noDecimal, 5)).toEqual('12345678900000');
+  expect(amountToDecimal(noDecimal, 10)).toEqual('1234567890000000000');
+  expect(amountToDecimal(noDecimal, 15)).toEqual('123456789000000000000000');
+  expect(amountToDecimal(noDecimal, 18)).toEqual('123456789000000000000000000');
+
+  const oneDecimal = '123456789.1';
+  expect(amountToDecimal(oneDecimal, undefined)).toEqual(undefined);
+  expect(amountToDecimal(oneDecimal, 0)).toEqual(undefined);
```

Comment:
I think it would be valid to drop precision, ie return `123456789` here. Not sure how much more difficult that would make the helpers?

Reply:
This is being done already, but in the form update method, so no change is needed